### PR TITLE
Update WireSlaveRequest.cpp

### LINKED ISSUE #5: Memory access violation on high data rates
Thank you for your library.
I used it in a project and stumbled upon an issue on high data rates.
I found two problems in the `WireSlaveRequest.cpp` where memory is accessed in an array without range checking the index.

I solved the problem by changing the line `while (unpacker.available())` in the `bool WireSlaveRequest::request(uint8_t address)` function to `while ((unpacker.available()) && (rxIndex_ < UNPACKER_BUFFER_LENGTH))`
and line `if (lastStatus_ == PACKET_READ && rxIndex_ < rxLength_)` in the `int WireSlaveRequest::read()` function to `if (lastStatus_ == PACKET_READ && rxIndex_ < rxLength_ && (rxIndex_ < UNPACKER_BUFFER_LENGTH))`

I will create a pull request with this fix.

### PR DESCRIPTION
**Pull request by TylerBird**
add boundary check for array index
fixes #5

### CODE REVIEW

**Review by gutierrezps — CHANGES_REQUESTED** [review]
Nice catch! But the change at line 107 is redundant, since `rxLength_` is set on line 76, and therefore is never greater than `UNPACKER_BUFFER_LENGTH` because of the fix you made at line 71.

Also, there are extra parenthesis at line 71. `while (unpacker.available() && rxIndex_ < UNPACKER_BUFFER_LENGTH)` does the trick.

File: src/WireSlaveRequest.cpp [hunk 1]
```diff
@@ -104,7 +104,7 @@ size_t WireSlaveRequest::available()
 int WireSlaveRequest::read()
 {
     int value = -1;
-    if (lastStatus_ == PACKET_READ && rxIndex_ < rxLength_) {
+    if (lastStatus_ == PACKET_READ && rxIndex_ < rxLength_ && (rxIndex_ < UNPACKER_BUFFER_LENGTH)) {
```
Comment (gutierrezps) [inline src/WireSlaveRequest.cpp]: Redundant check, can be how it was previously.

File: src/WireSlaveRequest.cpp [hunk 2]
```diff
@@ -68,7 +68,7 @@ bool WireSlaveRequest::request(uint8_t address)
 
     // copy payload bytes to rxBuffer
     rxIndex_ = 0;
-    while (unpacker.available()) {
+    while ((unpacker.available()) && (rxIndex_ < UNPACKER_BUFFER_LENGTH)) {
```
Comment (gutierrezps) [inline src/WireSlaveRequest.cpp]: Extra parentheses. Just write

`while (unpacker.available() && rxIndex_ < UNPACKER_BUFFER_LENGTH)`

**Review by gutierrezps — COMMENTED** [review]
Now I realized I could comment right on the lines.

File: src/WireSlaveRequest.cpp (hunk 2 above)
Comment (TylerBird) [reply src/WireSlaveRequest.cpp]: Yes of course. I would suggest that. 

```suggestion
    while (unpacker.available() && (rxIndex_ < UNPACKER_BUFFER_LENGTH)) {
```

File: src/WireSlaveRequest.cpp (hunk 1 above)
Comment (TylerBird) [reply src/WireSlaveRequest.cpp]:
```suggestion
    if (lastStatus_ == PACKET_READ && rxIndex_ < rxLength_) {
```

File: src/WireSlaveRequest.cpp (hunk 2 above)
Comment (TylerBird) [reply src/WireSlaveRequest.cpp]: Here also you can commit my suggestion and after that merge it to your master, if you like ;)

Otherwise nice project.

**Comment by gutierrezps** [comment]
@TylerBird I guess I don't have write permissions to commit your suggestion, since it's in your repo.

image

**Comment by TylerBird** [comment]
Big sorry again for letting waiting you so long. Now it should be fine, isnt it?
are you able to merge?

**Comment by gutierrezps** [comment]
I have just realized you made a pull request to your own repo 😆. You should do a pull request on my repo.

**Comment by TylerBird** [comment]
oh damn it.. yea that explains a lot ^^ so sorry, I have done that before, but not that bad 🗡️

**Comment by TylerBird** [comment]
I cant find a way to change the pull request target to your master. can u check your repo settings, if other peoples are allowed to contribute?

**Comment by TylerBird** [comment]
Any way, I made a new merge request to your master 👍  see https://github.com/gutierrezps/ESP32_I2C_Slave/pull/6

**Pull request merged**